src/ir/module-utils.cpp

```diff
@@ -692,6 +693,26 @@ std::vector<HeapType> getPublicHeapTypes(Module& wasm) {
     WASM_UNREACHABLE("unexpected export kind");
   }
 
+  // ConfigureAll in a start function makes its functions callable. They are
+  // only signature-called, so the heap type does not need to be public - nor
+  // types referred to - but for now we mark them as public to avoid breakage in
+  // several passes.
+  // TODO Specific fixes in those passes could replace this, and allow better
+  //      optimization.
+  if (wasm.start) {
```

Comment:
I imagine that eventually we'll need to handle this in places other than the start function, but starting with this is probably sufficient.